### ISSUE TITLE
Installation on windows is missing pkg_resources info

### ISSUE THREAD

**Issue by reszelaz** [conda-forge/pytango-feedstock#6]
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

PyTango modules are properly installed but the package is missing `pkg_resources` info:

```
>>> import tango
>>> import pkg_resources
>>> pkg_resources.get_distribution("PyTango")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\ProgramData\Miniconda3\envs\sardana\lib\site-packages\pkg_resources\__init__.py", line 478, in get_distribution
    dist = get_provider(dist)
  File "C:\ProgramData\Miniconda3\envs\sardana\lib\site-packages\pkg_resources\__init__.py", line 354, in get_provider
    return working_set.find(moduleOrReq) or require(str(moduleOrReq))[0]
  File "C:\ProgramData\Miniconda3\envs\sardana\lib\site-packages\pkg_resources\__init__.py", line 909, in require
    needed = self.resolve(parse_requirements(requirements))
  File "C:\ProgramData\Miniconda3\envs\sardana\lib\site-packages\pkg_resources\__init__.py", line 795, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'PyTango' distribution was not found and is required by the application
```

This does not happen when package is installed with `pip` (here on python 3.8):

```
>>> import pkg_resources
>>> pkg_resources.get_distribution("pytango")
pytango 9.3.4 (c:\programdata\miniconda3\envs\sardana-py3.8\lib\site-packages)
```

By copying the info dir installed by pip:
```
C:\ProgramData\Miniconda3\envs\sardana-py3.8\Lib\site-packages\pytango-9.3.4.dist-info
```
into:
```
C:\ProgramData\Miniconda3\envs\sardana\Lib\site-packages\pytango-9.3.4.dist-info
```

the issue seems to be solved.

### Installed packages

```shell
(sardana) C:\Users\vagrant>conda list
# packages in environment at C:\ProgramData\Miniconda3\envs\sardana:
#
# Name                    Version                   Build  Channel
asttokens                 2.0.5              pyhd8ed1ab_0    conda-forge
atomicwrites              1.4.1              pyhd8ed1ab_0    conda-forge
attrs                     22.1.0             pyh71513ae_1    conda-forge
backcall                  0.2.0              pyh9f0ad1d_0    conda-forge
backports                 1.0                        py_2    conda-forge
backports.functools_lru_cache 1.6.4              pyhd8ed1ab_0    conda-forge
boost                     1.74.0           py39hefe7e4c_5    conda-forge
boost-cpp                 1.74.0               h9f4b32c_8    conda-forge
brotli                    1.0.9                h8ffe710_7    conda-forge
brotli-bin                1.0.9                h8ffe710_7    conda-forge
bzip2                     1.0.8                h8ffe710_4    conda-forge
ca-certificates           2022.6.15            h5b45459_0    conda-forge
cached-property           1.5.2                hd8ed1ab_1    conda-forge
cached_property           1.5.2              pyha770c72_1    conda-forge
certifi                   2022.6.15        py39hcbf5309_0    conda-forge
click                     8.1.3            py39hcbf5309_0    conda-forge
colorama                  0.4.5              pyhd8ed1ab_0    conda-forge
cpptango                  9.3.5                h04899b1_1    conda-forge
cycler                    0.11.0             pyhd8ed1ab_0    conda-forge
debugpy                   1.6.0            py39h415ef7b_0    conda-forge
decorator                 5.1.1              pyhd8ed1ab_0    conda-forge
entrypoints               0.4                pyhd8ed1ab_0    conda-forge
executing                 0.9.1              pyhd8ed1ab_0    conda-forge
fisx                      1.2.0            py39h2e25243_2    conda-forge
fonttools                 4.34.4           py39hb82d6ee_0    conda-forge
freetype                  2.10.4               h546665d_1    conda-forge
future                    0.18.2           py39hcbf5309_5    conda-forge
guidata                   1.7.9              pyh9f0ad1d_0    conda-forge
guiqwt                    3.0.7            py39h79958ce_1    conda-forge
h5py                      3.7.0           nompi_py39hd4deaf1_100    conda-forge
hdf5                      1.12.1          nompi_h2a0e4a3_104    conda-forge
icu                       68.2                 h0e60522_0    conda-forge
iniconfig                 1.1.1              pyh9f0ad1d_0    conda-forge
intel-openmp              2022.1.0          h57928b3_3787    conda-forge
ipykernel                 6.15.1             pyh025b116_0    conda-forge
ipython                   8.4.0            py39hcbf5309_0    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
itango                    0.1.9              pyhd8ed1ab_0    conda-forge
jedi                      0.18.1           py39hcbf5309_1    conda-forge
jpeg                      9e                   h8ffe710_2    conda-forge
jupyter_client            7.3.4              pyhd8ed1ab_0    conda-forge
jupyter_core              4.11.1           py39hcbf5309_0    conda-forge
kiwisolver                1.4.4            py39h2e07f2f_0    conda-forge
krb5                      1.19.3               h1176d77_0    conda-forge
lcms2                     2.12                 h2a16943_0    conda-forge
lerc                      4.0.0                h63175ca_0    conda-forge
libblas                   3.9.0              15_win64_mkl    conda-forge
libbrotlicommon           1.0.9                h8ffe710_7    conda-forge
libbrotlidec              1.0.9                h8ffe710_7    conda-forge
libbrotlienc              1.0.9                h8ffe710_7    conda-forge
libcblas                  3.9.0              15_win64_mkl    conda-forge
libclang                  11.1.0          default_h5c34c98_1    conda-forge
libcurl                   7.83.1               h789b8ee_0    conda-forge
libdeflate                1.12                 h8ffe710_0    conda-forge
libffi                    3.4.2                h8ffe710_5    conda-forge
libiconv                  1.16                 he774522_0    conda-forge
liblapack                 3.9.0              15_win64_mkl    conda-forge
libpng                    1.6.37               h1d00b33_3    conda-forge
libsodium                 1.0.18               h8d14728_1    conda-forge
libssh2                   1.10.0               h680486a_2    conda-forge
libtiff                   4.4.0                ha17eb64_2    conda-forge
libwebp                   1.2.3                h8ffe710_1    conda-forge
libwebp-base              1.2.3                h8ffe710_2    conda-forge
libxcb                    1.13              hcd874cb_1004    conda-forge
libxml2                   2.9.14               hf5bbc77_3    conda-forge
libxslt                   1.1.35               h34f844d_0    conda-forge
libzlib                   1.2.12               h8ffe710_2    conda-forge
lxml                      4.9.1            py39hb82d6ee_0    conda-forge
lz4-c                     1.9.3                h8ffe710_1    conda-forge
m2w64-gcc-libgfortran     5.3.0                         6    conda-forge
m2w64-gcc-libs            5.3.0                         7    conda-forge
m2w64-gcc-libs-core       5.3.0                         7    conda-forge
m2w64-gmp                 6.1.0                         2    conda-forge
m2w64-libwinpthread-git   5.0.0.4634.697f757               2    conda-forge
matplotlib                3.5.2            py39hcbf5309_1    conda-forge
matplotlib-base           3.5.2            py39h581301d_1    conda-forge
matplotlib-inline         0.1.3              pyhd8ed1ab_0    conda-forge
mkl                       2022.1.0           h6a75c08_874    conda-forge
msys2-conda-epoch         20160418                      1    conda-forge
munkres                   1.1.4              pyh9f0ad1d_0    conda-forge
nest-asyncio              1.5.5              pyhd8ed1ab_0    conda-forge
numpy                     1.23.1           py39h1a62c8c_0    conda-forge
omniorb-libs              4.2.5                h8b1b97a_2    conda-forge
openjpeg                  2.4.0                hb211442_1    conda-forge
openssl                   1.1.1q               h8ffe710_0    conda-forge
packaging                 21.3               pyhd8ed1ab_0    conda-forge
parso                     0.8.3              pyhd8ed1ab_0    conda-forge
pickleshare               0.7.5                   py_1003    conda-forge
pillow                    9.2.0            py39ha53f419_0    conda-forge
pint                      0.19.2             pyhd8ed1ab_0    conda-forge
pip                       22.2.1             pyhd8ed1ab_0    conda-forge
pluggy                    1.0.0            py39hcbf5309_3    conda-forge
ply                       3.11                       py_1    conda-forge
prompt-toolkit            3.0.30             pyha770c72_0    conda-forge
psutil                    5.9.1            py39hb82d6ee_0    conda-forge
pthread-stubs             0.4               hcd874cb_1001    conda-forge
pthreads-win32            2.9.1                hfa6e2cd_3    conda-forge
pure_eval                 0.2.2              pyhd8ed1ab_0    conda-forge
py                        1.11.0             pyh6c4a22f_0    conda-forge
pygments                  2.12.0             pyhd8ed1ab_0    conda-forge
pymca                     5.7.4            py39h5d4886f_0    conda-forge
pyparsing                 3.0.9              pyhd8ed1ab_0    conda-forge
pyqt                      5.12.3           py39hb0d2dfa_4    conda-forge
pyqt5-sip                 4.19.18                  pypi_0    pypi
pyqtchart                 5.12                     pypi_0    pypi
pyqtgraph                 0.12.4             pyhd8ed1ab_0    conda-forge
pyqtwebengine             5.12.1                   pypi_0    pypi
pytango                   9.3.4            py39habe816b_0    conda-forge
pytest                    7.1.2            py39hcbf5309_0    conda-forge
python                    3.9.13          h9a09f29_0_cpython    conda-forge
python-dateutil           2.8.2              pyhd8ed1ab_0    conda-forge
python_abi                3.9                      2_cp39    conda-forge
pythonqwt                 0.8.3              pyhd8ed1ab_0    conda-forge
pywin32                   303              py39hb82d6ee_0    conda-forge
pyzmq                     23.2.0           py39he46f08e_0    conda-forge
qt                        5.12.9               h5909a2a_4    conda-forge
qtconsole                 5.3.1              pyhd8ed1ab_0    conda-forge
qtconsole-base            5.3.1              pyha770c72_0    conda-forge
qtpy                      2.1.0              pyhd8ed1ab_0    conda-forge
sardana                   3.3.3                    pypi_0    pypi
scipy                     1.9.0            py39h316f440_0    conda-forge
setuptools                63.3.0           py39hcbf5309_0    conda-forge
six                       1.16.0             pyh6c4a22f_0    conda-forge
sqlite                    3.39.2               h8ffe710_0    conda-forge
stack_data                0.3.0              pyhd8ed1ab_0    conda-forge
taurus                    5.1.4                hd8ed1ab_0    conda-forge
taurus-core               5.1.4              pyhd8ed1ab_0    conda-forge
taurus-qt                 5.1.4              pyhd8ed1ab_0    conda-forge
tbb                       2021.5.0             h2d74725_1    conda-forge
tgconf-mygui              0.1.0                    pypi_0    pypi
tk                        8.6.12               h8ffe710_0    conda-forge
tomli                     2.0.1              pyhd8ed1ab_0    conda-forge
tornado                   6.2              py39hb82d6ee_0    conda-forge
traitlets                 5.3.0              pyhd8ed1ab_0    conda-forge
tzdata                    2022a                h191b570_0    conda-forge
ucrt                      10.0.20348.0         h57928b3_0    conda-forge
unicodedata2              14.0.0           py39hb82d6ee_1    conda-forge
vc                        14.2                 hb210afc_6    conda-forge
vs2015_runtime            14.29.30037          h902a5da_6    conda-forge
wcwidth                   0.2.5              pyh9f0ad1d_2    conda-forge
wheel                     0.37.1             pyhd8ed1ab_0    conda-forge
xorg-libxau               1.0.9                hcd874cb_0    conda-forge
xorg-libxdmcp             1.1.3                hcd874cb_0    conda-forge
xz                        5.2.5                h62dcd97_1    conda-forge
zeromq                    4.3.4                h0e60522_1    conda-forge
zlib                      1.2.12               h8ffe710_2    conda-forge
zstd                      1.5.2                h6255e5f_2    conda-forge
```
```


### Environment info

```shell
(sardana) C:\Users\vagrant>conda info

     active environment : sardana
    active env location : C:\ProgramData\Miniconda3\envs\sardana
            shell level : 2
       user config file : C:\Users\vagrant\.condarc
 populated config files :
          conda version : 4.12.0
    conda-build version : not installed
         python version : 3.9.12.final.0
       virtual packages : __win=0=0
                          __archspec=1=x86_64
       base environment : C:\ProgramData\Miniconda3  (writable)
      conda av data dir : C:\ProgramData\Miniconda3\etc\conda
  conda av metadata url : None
           channel URLs : https://repo.anaconda.com/pkgs/main/win-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/win-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/msys2/win-64
                          https://repo.anaconda.com/pkgs/msys2/noarch
          package cache : C:\ProgramData\Miniconda3\pkgs
                          C:\Users\vagrant\.conda\pkgs
                          C:\Users\vagrant\AppData\Local\conda\conda\pkgs
       envs directories : C:\ProgramData\Miniconda3\envs
                          C:\Users\vagrant\.conda\envs
                          C:\Users\vagrant\AppData\Local\conda\conda\envs
               platform : win-64
             user-agent : conda/4.12.0 requests/2.27.1 CPython/3.9.12 Windows/10 Windows/10.0.19044
          administrator : True
             netrc file : None
           offline mode : False
```
```

**Comment by beenje** [comment]
Thanks for reporting this!

I had a look. On Windows, we build the C extension with cmake and then call a small script (`winsetup.py`) that runs `python  setup.py` with `build_py` and  `install_lib` (I patch that file in the conda recipe as we don't want to build a wheel). This doesn't create the metadata... Hence the issue you found.
I'm not sure why it was done like that in the first place. On Linux, we run `pip install` and metadata are properly created.

I guess that `bdist_wheel` does create the metadata and that's why the windows wheel works properly.

Running `python setup.py` is deprecated.
I can maybe patch something in the recipe but it would be better to improve the pytango build. I'll create an issue and try to find a solution.

**Comment by beenje** [comment]
pytango issue: https://gitlab.com/tango-controls/pytango/-/issues/460

**Comment by beenje** [comment]
This should be fixed in pytango 9.3.5 conda package.
I just tested it successfully.

Please let me know if you see issues.
